File: ruby/easy/excel_sheet_column_title.rb
# @param {Integer} n
# @return {String}
def convert_to_title(n)
    return ('A'.ord + n - 1).chr if n <= 26
    mod = (n % 26)
    div = (mod == 0) ? ((n/26).to_i - 1) : (n / 26).to_i
    return convert_to_title(div) + ((mod == 0) ? 'Z' : ('A'.ord + mod - 1).chr)
end

File: javascript/easy/excelSheetColumnTitle.js
/**
 * @param {number} n
 * @return {string}
 */
let aToZ = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
var convertToTitle = function(n) {
    if(n <= 26) {
        return aToZ[n-1];
    } else {
        let mod = (n % 26);
        let div = parseInt(n/26);
        return (mod == 0) ? (convertToTitle(div - 1) + "Z") : (convertToTitle(div) + (aToZ[mod-1]));
    }
};
